File: lc-blind75/ContainsDuplicate(lc-217)/Optimal/ContainsDuplicate.java
//tc:O(nlogn) sc:O(1)
import java.util.*;
public class ContainsDuplicate {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        int nums[]=new int [n];
        for(int i=0;i<n;i++){
            nums[i]=sc.nextInt();
        }
        System.out.println(findDuplicate(nums));
        sc.close();
    }
    public static boolean findDuplicate(int nums[]){
        int n=nums.length;
        boolean isDuplicate=false;
        Arrays.sort(nums);
        for(int i=0;i<n-1;i++){
            if(nums[i]==nums[i+1]){
                isDuplicate=true;
                break;
            }
        }
        return isDuplicate;
    }
}

File: lc-blind75/ContainsDuplicate(lc-217)/Optimal/ContainsDuplicate.py
#tc:O(nlogn) sc:O(1)
def findDuplicate(nums):
    n=len(nums)
    isDuplicate=False
    nums.sort()
    for i in range(n-1):
        if nums[i]==nums[i+1]:
            isDuplicate=True
            break
    return isDuplicate

n=int(input())
nums=list(map(int,input().split()))
print(findDuplicate(nums))